packages/trpc/server/routers/viewer/teams.tsx

```diff
@@ -562,4 +616,217 @@ export const viewerTeamsRouter = createProtectedRouter()
+        }
+      }
+
+      // Check to that subscription quantity is the same
+      if (subscriptionId || customer) {
+        // Grab current "incomplete" subscription
+        const subscriptionQuery = await retrieveTeamSubscription({
+          subscriptionId: subscriptionId,
+          customerId: customer.id,
+        });
+
+        if (subscriptionQuery && seats !== subscriptionQuery?.quantity) {
+          /* If the number of seats changed we need to cancel the current 
+          incomplete subscription and create a new one */
+          await deleteTeamSubscriptionQuantity(subscriptionId || subscriptionQuery.id);
+
+          const subscription = await createTeamSubscription(
+            customerId || customer.id,
+            input.billingFrequency,
+            input.seats
+          );
+
+          return {
+            clientSecret: subscription?.latest_invoice?.payment_intent?.client_secret,
+            customerId: customer.id,
+            subscriptionId: subscription.id,
+          };
+        }
+
+        // If customer exists and no changes were made to the subscription
+        if (subscriptionQuery)
+          return {
+            clientSecret: subscriptionQuery?.latest_invoice?.payment_intent?.client_secret,
+            customerId: customer.id,
+            subscriptionId: subscriptionQuery.id,
+          };
+      }
+
+      // If no changes then do not create a new customer & subscription, just return
+      if (!customerId && !subscriptionId) {
+        // First create the customer if it does not exist
+        customer = customer || (await createTeamCustomer(teamName, ctx.user.email));
+
+        // Create the subscription for the team
+        const subscription = await createTeamSubscription(customer.id, billingFrequency, seats);
+
+        // We just need the client secret for the payment intent
+        return {
+          clientSecret: subscription?.latest_invoice?.payment_intent?.client_secret,
+          customerId: customer.id,
+          subscriptionId: subscription.id,
+        };
+      }
+    },
+  })
+  .mutation("createTeam", {
+    input: z.object({
+      name: z.string(),
+      slug: z.string(),
+      logo: z.string().optional(),
+      members: z.array(
+        z.object({
+          name: z.union([z.string(), z.null()]),
+          email: z.string(),
+          id: z.number().optional(),
+          username: z.union([z.string(), z.null()]),
+          role: z.union([z.literal("OWNER"), z.literal("ADMIN"), z.literal("MEMBER")]),
+          avatar: z.union([z.string(), z.null()]),
+          locale: z.string(),
+        })
+      ),
+      customerId: z.string(),
+      subscriptionId: z.string(),
+    }),
+    async resolve({ ctx, input }) {
+      const { name, slug, logo, members, customerId, subscriptionId } = input;
+
+      if (!customerId && !subscriptionId)
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "Stripe Ids not found",
+        });
```

Comment:
Creating the team occurs after a successful payment so these ids should be present.